### PR TITLE
None for rational(Float) query

### PR DESCRIPTION
**Pull request by abhigyank**
#### References to other Issues or PRs

Fixes issue #12168 

#### Brief description of what is fixed or changed
A ```CommonHandler``` function  with ```@staticmethod``` returning ```None``` for ```Float``` values.
Changed existing Float tests for rational to assert for ```None``` instead of ```False```
#### Other comments
```
 >>> ask(Q.rational(math.pi))
 None
```

### CODE REVIEW

File: sympy/assumptions/handlers/common.py
Comment (gxyd) [inline sympy/assumptions/handlers/common.py]: I don't think this would be needed.

File: sympy/assumptions/handlers/sets.py
Comment (gxyd) [inline sympy/assumptions/handlers/sets.py]: The routine you created above in handlers/sets.py should be here, instead of assigning `Float` like this.

File: sympy/assumptions/tests/test_query.py
Comment (gxyd) [inline sympy/assumptions/tests/test_query.py]: I wouldn't create a separate functions, just add the test for `ask(Q.rational(math.pi))` with the others above with a comment "# test for issue #12168".

**Comment by gxyd** [comment]
Don't use 'Fixes issue #12168' use 'Fixes #12168' to actually make use of github's auto-closing syntax.

**Comment by gxyd** [comment]
The current way just works. But I think having a common `AlwaysNone` staticmethod in `CommonHandler` class with `Float = staticmethod(CommonHandler.AlwaysNone)` instead what I suggested earlier might be better.

**Comment by gxyd** [comment]
WDYT?

**Comment by abhigyank** [comment]
That could also be done. I too feel that is a better way of doing as well. In future,  ```AlwayNone``` method can also be used from ```CommonHandler``` for other purposes as well. I'll do make another commit making this change and push it?

**Comment by gxyd** [comment]
Yes, please do so.

**Review by gxyd — APPROVED** [review]
LGTM. Let's wait for another review from someone.

**Comment by abhigyank** [comment]
ping @asmeurer

File: sympy/assumptions/tests/test_query.py
Comment (asmeurer) [inline sympy/assumptions/tests/test_query.py]: This test is unnecessary. `z` is already a float tested above.

**Comment by gxyd** [comment]
Also, I just saw this comment by @jksuom https://github.com/sympy/sympy/pull/12170#issuecomment-279489529 
Quoting here:
> It is probably overkill to define an AlwaysNone handler as it seems not to be used elsewhere. It would suffice to write an ordinary staticmethod def Float returning None.
> The comment about finite precision is irrelevant in either case.
> Some tests should be fixed to satisfy the changed results. You can find them locally by running bin/test query and bin/doctest.

Still, I'm more inclined towards having `AlwaysNone` method in `CommonHandler`, instead of the other suggestion (which was done earlier in this PR as well).

**Comment by jksuom** [comment]
I'm not opposed to having `AlmostNone` (though it might currently not be used elsewhere).

**Comment by abhigyank** [comment]
I think we can have ```AlwaysNone``` method in ```CommonHandler``` as @gxyd suggest. Even though it might not be used currently elsewhere as @jksuom says, I feel it can be used in the future somewhere as it is not uncommon to come up with such situations.

**Comment by gxyd** [comment]
Yes. Thanks @jksuom for clarifying :)

I'm merging this.

**Pull request merged**
